e2e_test/batch/functions/date_bin.part

```diff
@@ -0,0 +1,44 @@
+query T
+SELECT date_bin(
+  '2 days'::interval,
+  '2021-01-02 12:00:00+08'::TIMESTAMPTZ,
+  '2021-01-01 00:00:00+00'::TIMESTAMPTZ
+);
+----
+2021-01-01 00:00:00+00:00
+
+query T
+SELECT date_bin(
+  '3 hours'::interval,
+  '2021-01-01 05:45:00'::TIMESTAMP,
+  '2021-01-01 00:00:00'::TIMESTAMP
+);
+----
+2021-01-01 03:00:00
+
+query T
+SELECT date_bin(
+  '30 minutes'::interval,
+  '2021-01-01 02:17:00-05'::TIMESTAMPTZ,
+  '2021-01-01 01:00:00-05'::TIMESTAMPTZ
+);
+----
+2021-01-01 07:00:00+00:00
+
+query T
+SELECT date_bin(
+  '1 hour'::interval,
+  '2021-06-15 13:30:00'::TIMESTAMP,
+  '2021-06-15 00:00:00+02'::TIMESTAMPTZ
+);
+----
+2021-06-15 13:00:00+00:00
+
+query T
+SELECT date_bin(
+  '15 minutes'::interval,
+  '2021-12-31 23:59:59+00'::TIMESTAMPTZ,
+  '2021-12-31 22:00:00'::TIMESTAMP
+);
+----
+2021-12-31 23:45:00+00:00
```

Comment:
Some alternative test cases to demonstrate the expected behavior of timezone handling:
```
set timezone = 'US/Pacific';

select date_bin(interval '1' day, '2021-09-08T18:05:04-07:00'::timestamptz, '2021-01-01'::timestamp);
----
2021-09-08 01:00:00-07:00

set timezone = 'UTC';
```

This case actually tests 2 things:
* `'2021-01-01'::timestamp` is interpreted in session time zone `US/Pacific` thus `1609488000000000` (rather than in UTC `1609459200000000`)
* `1 day` got no special treatment regarding DST. It is always 24 hours instead of 23 or 25. So it is NOT `2021-09-08 00:00:00-07:00` but `2021-09-08 01:00:00-07:00`

Reply:
Added. Thanks for the test case!